### PR TITLE
LF-4306 No inline error for character count for describe the purpose

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This PR adds/fixes the max character validation and inline validation error for the three text input fields. It does not do the React Select or the `trim()` validation, just the max character on:
1. `other_purpose` (this was the topic of both linked Jira tickets below)
2. `other_application_method` -- no Jira ticket for this one but field has same db datatype and would give same 400 error
3. `supplier` -- no Jira ticket for this one and had a maxLength, so form would be invalid and no data would get to the database, but it didn't use `hookFormMaxCharsValidation` so there was no inline error

I actually thought one of the other fields had a bug table record of its own but maybe we didn't give it blocker priority

Jira links:
https://lite-farm.atlassian.net/browse/LF-4306
https://lite-farm.atlassian.net/browse/LF-4323

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by SayakaOno — APPROVED** [review]
Thank you!

**Comment by Duncan-Brain** [comment]
@kathyavini is this an appropriate place to also fix the model validation similar to how it is on cropModel fro the same properties?

**Comment by kathyavini** [comment]
@Duncan-Brain done! Models added.

File: packages/api/src/models/productModel.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ class ProductModel extends baseModel {
         product_id: { type: 'integer' },
         name: { type: 'string' },
         product_translation_key: { type: 'string' },
-        supplier: { type: 'string' },
+        supplier: { type: 'string', minLength: 1, maxLength: 255 },
```
Comment (SayakaOno) [inline packages/api/src/models/productModel.js:39]: I get `"error":{"message":"supplier: must NOT have fewer than 1 characters"}` for a farm not pursuing a certification!

Comment (kathyavini) [reply packages/api/src/models/productModel.js]: Oooh thank you for spotting that @SayakaOno!!

I added `null` to the types but noticed that doesn't stop triggering the database error, so removed `minLength` too. I wonder why the other two fields work with just including `null`! 🤔🤔🤔

Comment (Duncan-Brain) [reply packages/api/src/models/productModel.js]: @kathyavini I think we check and trim string on backend for product so an empty string is maybe supplied. (Or to handle api case). I can't see where the model is checked before `checkProductValidity()` though.

**Review by Duncan-Brain — APPROVED** [review]

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**